### PR TITLE
mempool sync update

### PR DESCRIPTION
**Pull request by GusevTimofey**

### CODE REVIEW

File: ergo-chain-sync/src/lib.rs [hunk 1]
```diff
@@ -56,19 +57,25 @@ impl<'a, TClient, TCache> ChainSyncNonInit<'a, TClient, TCache> {
...
         starting_height: u32,
         tip_reached_signal: Option<&'a Once>,
-    ) -> ChainSync<'a, TClient, TCache> {
-        ChainSync::init(starting_height, self.client, self.cache, tip_reached_signal).await
+    ) -> Pin<Box<dyn Stream<Item=ChainUpgrade> + 'a>> {
+        Box::pin(chain_sync_stream(
+            ChainSync::init(starting_height,
+                            self.client,
+                            self.cache,
+                            tip_reached_signal).await
```
Comment (oskin1) [inline ergo-chain-sync/src/lib.rs:76]: Please use rustfmt, it seems it should not be formatted this way.

File: ergo-mempool-sync/src/lib.rs [hunk 2]
```diff
@@ -62,27 +71,29 @@ impl SyncState {
...
 }
 
 #[pin_project::pin_project]
-pub struct MempoolSync<'a, TClient, TChainSync> {
+pub struct MempoolSync<'a, TClient> {
     conf: MempoolSyncConf,
     client: &'a TClient,
     #[pin]
-    chain_sync: TChainSync,
+    chain_sync: Pin<Box<dyn Stream<Item=ChainUpgrade> + 'a>>,
```
Comment (oskin1) [inline ergo-mempool-sync/src/lib.rs:82]: Why removing `TChainSync` abstraction?

File: ergo-mempool-sync/src/lib.rs [hunk 3]
```diff
@@ -156,49 +167,68 @@ async fn sync<'a, TClient: ErgoNetwork>(client: &TClient, mut state: RefMut<'a,
...
 }
 
-impl<'a, TClient, TChainSync> Stream for MempoolSync<'a, TClient, TChainSync>
-    where
-        TClient: ErgoNetwork,
-        TChainSync: Stream<Item=ChainUpgrade> + Unpin,
+pub enum Combined {
+    Mempool(MempoolUpdate),
+    Empty,
+}
```
Comment (oskin1) [inline ergo-mempool-sync/src/lib.rs:173]: `Option<MempoolUpdate>`?

Comment (GusevTimofey) [reply ergo-mempool-sync/src/lib.rs:173]: I'm not sure, the reference is: https://github.com/spectrum-finance/spectrum/blob/1c03dcd9e643211c70990a683dffa22520b109ef/spectrum-network/tests/integration_tests/mod.rs#L328-L334

**Pull request closed without merging**